### ISSUE TITLE
Stats edits seem to save without clicking Save button

### ISSUE THREAD

**Issue by pashiav** [specify/specify7#4735]
**Describe the bug**
When Editing Statistics, any changes made seem to be Saved without clicking the Save button when navigating away then back. The user should be prompted with a dialog asking if they want to save their changes. 

Note: refreshing the page shows that the changes were not truly saved, but this is confusing to users

**To Reproduce**
1. Go to Statistics (https://umherb11024-edge.test.specifysystems.org/specify/stats)
2. Click on **Edit**
3. Make new changes (e.g. add categories)
4. Navigate away from Statistics
5. Go back to Statistics
6. See that changes made were saved on the page

**Expected behavior**
Edits should not be saved if user does not click the **Save** button. The user should be prompted with a warning dialog asking "Are you sure you want to leave this page?" when they try to navigate away after making unsaved changes. 

![Screen Shot 2024-04-03 at 12 13 42 PM](https://github.com/specify/specify7/assets/118556884/40e330a8-be29-4491-9682-423e58cc1499)

**Evidence**
`edge`
Page refresh starts 0:13 - skip to 1:01 to see page refreshed. I also noticed that it normally takes a while for the Statistics page to refresh when unsaved changes are made.

https://github.com/specify/specify7/assets/118556884/ce6996f3-9be7-4f03-8e7f-e61345717040

OS: macOS Monterey (12.6.5)
Browser: Chrome
Specify 7 Version: `edge`
Database Name: `umherb_1_10_24`
[Specify 7 System Information - 2024-04-03T16_58_07.483Z.txt](https://github.com/specify/specify7/files/14855838/Specify.7.System.Information.-.2024-04-03T16_58_07.483Z.txt)

**Comment by realVinayak** [comment]
Yeah, that's expected, since the save button is a fake save button, and it automatically saves after 5 seconds of making a change

**Comment by maxpatiiuk** [comment]
@realVinayak should we remove the "Are you sure you want to leave this page?" dialog in such case?
i.e we don't have this dialog in the preferences page

**Comment by emenslin** [comment]
Can recreate in edge (7.9.6)
